examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -425,11 +425,11 @@ CHIP_ERROR PairingCommand::UpdateNetworkAddress()
 void PairingCommand::OnAddressUpdateComplete(NodeId nodeId, CHIP_ERROR err)
 {
     ChipLogProgress(chipTool, "OnAddressUpdateComplete: %" PRIx64 ": %s", nodeId, ErrorStr(err));
-    if (err != CHIP_NO_ERROR && nodeId == mNodeId)
+    if (err != CHIP_NO_ERROR)
     {
-        // Set exit status only if the address update failed.
-        // Otherwise wait for OnCommissioningComplete() callback.
-        SetCommandExitStatus(err);
+        // For some devices, it may take more time to appear on the network and become discoverable
+        // over DNS-SD, so don't give up on failure and restart the address update.
+        UpdateNetworkAddress();
```

Comment:
how long will this take? Could we set some counter for retries or have different mDNS timeouts?

Reply:
All `chip-tool` commands have a timeout. In the case of `pairing` commands it is equal to 120s (hardcoded).

Reply:
A comment here saying why this is not a 'retry forever' would help. I am afraid of the code to not be copied and pasted in time and lead to some infinite retries.